### PR TITLE
Create ES secret and add to YAML to be applied on managed cluster

### PR DESCRIPTION
**Pull request by pfmackin**
# Description

1. When VMC is reconciled, create the Elasticsearch secret used by the managed cluster to ship logs to admin ES
2. Add the Elasticsearch secret to the YAML that will be applied on the managed cluster
3. Slight reorg of cluster controller tests to reduce the size of a function

Fixes VZ-2284

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/clusters/sync_es_secret.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	"github.com/verrazzano/verrazzano/platform-operator/constants"
+	k8net "k8s.io/api/networking/v1beta1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	caKey       = "ca.crt"
```
Comment (desagar) [inline platform-operator/controllers/clusters/sync_es_secret.go:19]: this key is supposed to be ca-bundle in the generated YAML. At least, that's the assumption made by the logging scope code.

Comment (pfmackin) [reply platform-operator/controllers/clusters/sync_es_secret.go:19]: we already use cr.crt in system.tls and should be consistent.  I can change the logging code

File: platform-operator/controllers/clusters/sync_manifest.go [hunk 2]
```diff
@@ -16,29 +16,43 @@ import (
...
 // Create or update a secret that contains Kubernetes resource YAML which will be applied
 // on the managed cluster to create resources needed there. The YAML has multiple Kubernetes
 // resources separated by 3 hyphens ( --- ), so that applying the YAML will create/update multiple
 // resources at once.  This YAML is stored in the Verrazzano manifest secret.
 func (r *VerrazzanoManagedClusterReconciler) syncManifestSecret(vmc *clusterapi.VerrazzanoManagedCluster) error {
 	const (
-		yamlSep                           = "---\n"
-		targetRegistrationSecretName      = "verrazzano-cluster"
-		targetRegistrationSecretNamespace = "verrazzano-system"
+		yamlSep                            = "---\n"
```
Comment (mgianatagh) [inline platform-operator/controllers/clusters/sync_manifest.go:30]: Can this line be removed?  there is a declaration of yamlSep on line 21

File: platform-operator/controllers/clusters/sync_manifest.go [hunk 3]
```diff
@@ -16,29 +16,43 @@ import (
...
+		targetRegistrationSecretName       = "verrazzano-cluster"
+		targetRegistrationSecretNamespace  = "verrazzano-system"
```
Comment (mgianatagh) [inline platform-operator/controllers/clusters/sync_manifest.go:32]: use the constant for "verrazzano-system", same for line 34

Comment (pfmackin) [reply platform-operator/controllers/clusters/sync_manifest.go]: fixed

File: platform-operator/controllers/clusters/sync_manifest.go (hunk 2 above)
Comment (pfmackin) [reply platform-operator/controllers/clusters/sync_manifest.go]: fixed

File: platform-operator/controllers/clusters/sync_es_secret.go (hunk 1 above)
Comment (pfmackin) [reply platform-operator/controllers/clusters/sync_es_secret.go:19]: I will change to ca-bundle for now

Comment (pfmackin) [reply platform-operator/controllers/clusters/sync_es_secret.go]: fixed

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**